test/test_site/index.md

```diff
@@ -12,7 +12,9 @@
 <include src="requirements/SpecifyingRequirements.md" name="Dynamic Include" dynamic />
 
 # Boilerplate include
-<include src="requirements/boilerTest.md" name="Boilerplate Referencing" />
+<include src="requirements/boilerTest.md" name="Boilerplate Referencing" boilerplate />
+
+<include src="requirements/notInside.md" name="Referencing specified path in boilerplate" boilerplate="folder/inside.md" dynamic/>
```

Comment:
a very noob question here:

So if I already know I want to use `folder/inside.md` here and not what's in `src`, do I have to write both `src` and `boilerplate` together? Can I just use something like:

`<include boilerplate="folder/inside.md" name="Boilerplate Referencing" />`

Remind me if this is already discussed when we design the `boilerplate` feature

Reply:
[An error node is returned](https://github.com/rachx/markbind/commit/ebeb9c21e840224fb0883beeacbb0dc734591fc4#diff-ba2dd3dbaf0965b1ae992e6f64d7109fR106) if `src` is empty.

`src` is used as [`asIfTo`](https://github.com/MarkBind/markbind/pull/167/commits/5aeb0eb387c6f9fb3442ca4ab4df6d403e7d27f7#diff-ba2dd3dbaf0965b1ae992e6f64d7109fR130) and to determine [`filePath`](https://github.com/MarkBind/markbind/pull/167/commits/b57b9628db8ddd1178445c9e993472bc8c86b113#diff-ba2dd3dbaf0965b1ae992e6f64d7109fR117)/[`asIfAt`](https://github.com/MarkBind/markbind/pull/167/commits/388458453b4653d5c76e97fa77b0f09fa70a3c6b#diff-ba2dd3dbaf0965b1ae992e6f64d7109fR54) for `calculateBoilerplateFilePath`.
I think we should avoid inferring, for your example, between:
- `<include src="inside.md" boilerplate="folder/inside.md" name="Boilerplate Referencing" />`
- `<include src="folder/inside.md" boilerplate="folder/inside.md" name="Boilerplate Referencing" />`

Reply:
`src` has to be written:

`src` is where it pretends to be. It is needed to set the context/current working file so that if the boilerplate file contains `<include src ="child.md">`, it will know which folder to look for `child.md` in.

Also, for dynamically include, the _include_.html files are created where they are supposed to be.